conf/hbase-site.xml

```diff
@@ -21,4 +20,31 @@
  */
 -->
 <configuration>
+  <!--
+    The following properties are set for running HBase as a single process on a
+    developer workstation. With this configuration, HBase is running in
+    "stand-alone" mode and without a distributed file system. In this mode, and
+    without further configuration, HBase and ZooKeeper data are stored on the
+    local filesystem, in a path under the value configured for `hbase.tmp.dir`.
+    This value is overridden from its default value of `/tmp` because many
+    systems clean `/tmp` on a regular basis. Instead, it points to a path within
+    this HBase installation directory.
+
+    Running against the `LocalFileSystem`, as opposed to a distributed
+    filesystem, runs the risk of data integrity issues and data loss. Normally
+    HBase will refuse to run in such an environment. Setting
+    `hbase.unsafe.stream.capability.enforce` to `false` overrides this behavior,
+    permitting operation. This configuration is for the developer workstation
+    only and __should not be used in production!__
```

Comment:
Is this copy of hbase-site.xml what makes it into the binary tarball? If so, does that worry you at all?

I'm lamenting that we don't have a good way to differentiate a "user doing things locally/in-situ" vs. "a user building out a real system". Eventually, we can't stop someone from shooting themselves in the foot...

Reply:
Yes, this is what gets shipped as `conf/hbase-site.xml`. Which part has you concerned? Now's the time to air all grievances :)

Reply:
I'm just trying to think of ways that we can better prevent a user from doing themselves harm, and ways that people could fall into trouble if this is in the distribution's default configuration file. e.g. I could imagine a process which starts with hbase-site.xml from a release tarball, and then additions on top of that. This change would adversely affect them.

The only other thing I could think of would be to carry `hbase.cluster.distributed=false` here to try to be more confident that they are actually changing the config, giving it some thought. I'm mostly just thinking out loud though. I'm not personally convinced that turning off the sync enforcement OOTB is "worrisome".

Reply:
> The only other thing I could think of would be to carry hbase.cluster.distributed=false here to try to be more confident that they are actually changing the config, giving it some thought. I'm mostly just thinking out loud though. I'm not personally convinced that turning off the sync enforcement OOTB is "worrisome".

I suppose it's better here than by default, as was done in the patch that this PR reverts.

IMHO, without a `LocalFIleSystem` that's actually safe to use, none of these solutions are very pleasant.

Reply:
> The only other thing I could think of would be to carry `hbase.cluster.distributed=false` here to try to be more confident that they are actually changing the config, giving it some thought.

This is the default value, but sure, I see no harm in adding it here explicitly. A user is expected to edit that value when they go to production anyway, so there's no harm in saving them some typing.

> I could imagine a process which starts with hbase-site.xml from a release tarball, and then additions on top of that. This change would adversely affect them.

You're correct, this change affects them. After reflection, I prefer the change where we ship these defaults configured for development/testing users, instead of the initial change that altered the code.